### PR TITLE
Add QoL helpers for building 'closures'

### PR DESCRIPTION
**Pull request by SimonBoothroyd**
## Description
Provide a brief description of the PR's purpose here.

## Status
- [ ] Ready to go

### CODE REVIEW

File: descent/targets/thermo.py [hunk 1]
```diff
@@ -631,10 +620,77 @@ def predict(
             torch.nan if entry["std"] is None else entry["std"] * abs(type_scale)
         )
 
+        if verbose:
+            std_ref = "" if entry["std"] is None else " ± {float(entry['std']):.3f}"
```
Comment (jthorton) [inline descent/targets/thermo.py:624]:
```suggestion
            std_ref = "" if entry["std"] is None else f" ± {float(entry['std']):.3f}"
```

File: descent/targets/thermo.py [hunk 2]
```diff
@@ -631,10 +620,77 @@ def predict(
...
+                {
+                    "type": f'{entry["type"]} [{entry["units"]}]',
+                    "smiles_a": descent.utils.molecule.unmap_smiles(entry["smiles_a"]),
+                    "smiles_b": (
+                        ""
+                        if entry["smiles_b"] is None
+                        else descent.utils.molecule.unmap_smiles(entry["smiles_b"])
+                    ),
+                    "pred": f"{float(value):.3f} ± {float(std):.3f}",
+                    "ref": f"{float(entry['value']):.3f}{std_ref}",
```
Comment (jthorton) [inline descent/targets/thermo.py:636]: If `entry["std"]` is `None` this prints the string `" ± {float(entry['std']):.3f}"`

File: descent/targets/thermo.py (hunk 1 above)
Comment (jthorton) [inline descent/targets/thermo.py:624]:
```suggestion
            std_ref = "" if entry["std"] is None else f" ± {float(entry['std']):.3f}"
```

Comment (SimonBoothroyd) [reply descent/targets/thermo.py]: good catch, should be fixed in the latest commit!

**Comment by SimonBoothroyd** [comment]
@jthorton i'm going to split the `train` helpers out into their own PR to make it a bit cleaner

**Pull request merged**